jacquard/odm/base.py

```diff
@@ -63,6 +63,11 @@ def __init__(self, pk, **fields):
         for field_name, value in fields.items():
             setattr(self, field_name, value)
 
+    @classmethod
+    def upgrade_data(cls, data):
```

Comment:
Are the semantics of _how old_ this `data` is allowed to be recorded somewhere? If not, please record that.

Reply:
I don't think I understand what this means

Reply:
From the perspective of someone implementing this method, it's not immediately clear what the possible values of `data` are. My initial thought was that an author might want to remove some historic migration code after a while, though I now realise that that won't ever be possible under the current design. (The answer therefore being "data can be from the beginning of time and newer).

However, this leads me to a larger question: how does the author know what format `data` is at all. Currently this seems to rely on knowing/inspecting the internals of the ODM classes, for all possible versions which a model might have been saved under. It would be good to document both what the `data` represents and what sorts of change cause it to need upgrading.

I'm not sure how whether models are expected to be something that non-core-developers (i.e: plugin writers and users) would ever need to think about, so it may be that this concern is minor. If so, then just a little more information in the docstring would be useful.

Reply:
In principle plugin writers and users should be able to use the ODM.

What do you reckon on making this an initial transitional mechanism? As in we'll work out the details of doing upgrades better in the future, but for now use this mechanism as-is, with a different name and documentation clarifying what it currently does?

Reply:
fbbb5ec003315ca0d5e9aa929f73d8e72019770c and 035f95d0ee738cfe8bd8adcd087339e846544dba for now.